images/dvcr-artifact/cmd/dvcr-cleaner/cmd/delete.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+var DeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete `VirtualImages` or `ClusterVirtualImages`",
+	Args:  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+}
+
+var deleteViCmd = &cobra.Command{
+	Use:   "vi",
+	Short: "Delete a `VirtualImage` or all `VirtualImages` in namespace",
+	Args: cobra.MatchAll(
+		cobra.MaximumNArgs(1),
+		cobra.OnlyValidArgs,
+	),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		imgsDir := fmt.Sprintf("%s/vi/%s", RepoDir, NamespaceFlag)
+		err := DeleteImage(virtv2.VirtualImageKind, imgsDir, cmd, args)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	SilenceUsage:  true,
+	SilenceErrors: true,
+}
+
+var deleteCviCmd = &cobra.Command{
+	Use:   "cvi",
+	Short: "Delete a `ClusterVirtualImage` or all `ClusterVirtualImages`",
+	Args: cobra.MatchAll(
+		cobra.MaximumNArgs(1),
+		cobra.OnlyValidArgs,
+	),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		imgsDir := fmt.Sprintf("%s/cvi", RepoDir)
+		err := DeleteImage(virtv2.ClusterVirtualImageKind, imgsDir, cmd, args)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	SilenceUsage:  true,
+	SilenceErrors: true,
+}
+
+func DeleteImage(imageType, imgsDir string, cmd *cobra.Command, args []string) error {
```

Comment:
don't used imagetype

Reply:
it should be used in `removeImageDir` func, thanks